pkg/build/builder/source.go

```diff
@@ -56,38 +65,52 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 	return nil
 }
 
+// checkRemoteGit validates the specified Git URL. It returns GitNotFoundError
+// when the remote repository not found and GitAuthenticationError when the
+// remote repository failed to authenticate.
+// Since this is calling the 'git' binary, the proxy settings should be
+// available for this command.
```

Comment:
@bparees I tested this manually with broken proxy (by accident) and it is working properly. We should have extended test that exercise this behavior. I can add one.